itensor/itensor_interface.ih

```diff
@@ -290,6 +379,21 @@ set(Cplx val)
         }
     }
 
+template<typename IndexT>
+void ITensorT<IndexT>::
+set(Real val)
```

Comment:
I initially didn't include it, I think it caused some errors in the unit tests without it. I will play around with removing it.

Reply:
I tried removing .set(Real) and the unit test A.set(4.5) failed. It seemed like it was dispatching on the new .set(int, ...) instead of .set(Cplx), which doesn't make sense to me, but I'll keep .set(Real) there for now.

Reply:
Oh, I see. That's very interesting. Yes these are some of the hairy situations one can get into with templates and automatic type conversions and things. It also raises a key question, which is: does the code behave correctly for the case where say I have a 0-index tensor but I want to set its value using an integer as the value? I.e. I want to do T.set(3); so that T's value is now 3. Let's make sure there is a unit test that checks this case is working properly. (I'll take a look too but we should both look.) If needed, we may have to fall back onto using some type traits but I hope it won't come to that!